aten/src/ATen/native/Batching.cpp

```diff
@@ -20,6 +21,66 @@ static bool has_level(const Tensor& self, int64_t level) {
   return it != bdims.end();
 }
 
+// Returns a Tensor with batch dim with level `level` turned into a regular tensor,
+// as well as a logical dim index of where said dimension is in the returned tensor.
```

Comment:
This is... kind of a weird thing to want to do.

Reply:
I can add in an additional comment about why we want to do this (I"ll just copy/paste the following)

The reason why we want to do this is because we want to keep track of which dimension used to be the batch dimension so that we can move it to the correct logical dimension. For example, if we had:
```
x = torch.randn(2, 3, 5)
vmap(lambda x: x, out_dims=1)(x)
```
then right when we are about to exist the vmap block, x is a BatchedTensor with a batch dimension at (physical) index 0. The batch dimension doesn't always have to exist at (physical) index 0.  When we undo the batch dimension, we want to move it to dimension 1 (as specified by `out_dims`). So we return the index at which the batch dim appears so that we can move it to the correct place.

Reply:
Hmm, I realize that the comment is wrong the way it is written and that that might be the source of the confusion. Let me fix.

Reply:
Is there ever a situation where you'd want to this operation without immediately moving the batch dimension afterwards?

Reply:
No, there isn't -- would it be easier to read if I tossed the `movedim` calls into this function?

Reply:
Not necessarily, the function is pretty long. The difficulty is distinguishing between whether or not this is a semantically meaningful operation, or if it's just a helper operation for some other semantically meaningful operation. C++ doesn't make it too easy to express your intent here.